src/Common/src/System/IO/RowConfigReader.cs

```diff
@@ -0,0 +1,117 @@
+﻿using System.Diagnostics;
+
+namespace System.IO
+{
+    /// <summary> 
+    /// Helper for reading config files where each row is a key-value data pair.
+    /// The input key-values must not have any whitespace within them.
+    /// </summary>
+    internal struct RowConfigReader
```

Comment:
1. To avoid unnecessary allocations, and as a parallel to the existing StringParser. It takes some care to use as it is a struct, but it's also generally just used as a local, so there's not a lot to be concerned about.

2. I'm currently writing some tests for the code that uses the RowConfigReader in NetworkInformation, but I will also write some individual tests for the type itself. I understand there's actually a test project for the Common directory that I can add to.